doc/api.js

```diff
@@ -0,0 +1,407 @@
+ * @param {Function} callback A callback function to handle the async response.
+ *   It shall expect two parameters: ({@link module:nrfjprogjs~Error|Error}, {@link module:nrfjprogjs~DeviceInformation|DeviceInformation}).
+ */
+export function getDeviceInfo(serialNumber, callback) {}
+
+
+
+/**
+ * Async function to read a chunk of memory. The data received by the callback
+ * is an array of integers, each of them representing a single byte (with values
+ * from 0 to 255).
+ * <br/>
+ *
+ * Please note that the data is an array of numbers - it is NOT a {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array|UInt8Array},
+ * and it is NOT a {@link https://nodejs.org/api/buffer.html|Buffer}.
+ * <br/>
+ *
+ * This is the same functionality as running "<tt>nrfjprog --memrd</tt>" in the command-line tools.
+ *
+ * @example
+ * nrfjprogjs.read(123456789, 0, 16, function(err, data) {
+ *      if (err) throw err;
+ *      console.log('The first 16 bytes of memory look like: ' + data.join(','));
+ * } );
+ *
+ * @param {integer} serialNumber The serial number of the device to query
+ * @param {integer} address The start address of the block of memory to be read
+ * @param {integer} length The amount of bytes to be read
+ * @param {Function} callback A callback function to handle the async response.
+ *   It shall expect two parameters: ({@link module:nrfjprogjs~Error|Error}, Array of integers).
+ */
+export function read(serialNumber, address, length, callback) {}
+
+
+
+// TODO: What is the endianness of this???
+/**
+ * Async function to read a single 4-byte word from memory.
+ * <br/>
+ *
+ * Please note that the data is a number - it is NOT a {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint32Array|UInt32Array},
+ * and it is NOT a {@link https://nodejs.org/api/buffer.html|Buffer}.
+ * <br/>
+ *
+ * This is the same functionality as running "<tt>nrfjprog --memrd -w</tt>" in the command-line tools.
+ *
+ * @example
+ * nrfjprogjs.read(123456789, 0, function(err, data) {
+ *      if (err) throw err;
+ *      console.log('The first word of memory looks like: ' + data);
+ * } );
+ *
+ * @param {integer} serialNumber The serial number of the device to query
+ * @param {integer} address The address of the word to be read
+ * @param {Function} callback A callback function to handle the async response.
+ *   It shall expect two parameters: ({@link module:nrfjprogjs~Error|Error}, integer).
+ */
+export function readU32(serialNumber, address, callback) {}
+
+
+
+/**
+ * Async function to push a program to the device.
+ * <br/>
+ *
+ * This is the same functionality as running "<tt>nrfjprog --program</tt>" in the command-line tools.
+ *
+ * @example
+ * nrfjprogjs.program(123456789, "/some/path/nrf52832_abcd.hex", {}, function(err) {
+ *      if (err) throw err;
+ * } );
+ *
+ * @param {integer} serialNumber The serial number of the device to query
```

Comment:
I think I copy-pasted that from the `getDeviceInfo()` docs, where the capabilities are in fact queried, but yeah, the wording is different for each function